tutorials/LinearAlgebra/LinearAlgebra.ipynb

```diff
@@ -0,0 +1,983 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:blue\">Exercise 2</span>: Scalar multiplication\n",
+    "\n",
+    "**Inputs:**\n",
+    "\n",
+    "1. A scalar $x$.\n",
+    "2. An $n \\times m$ matrix $A$.\n",
+    "\n",
+    "**Goal:** Return the matrix that results from the multiplication $x \\cdot A$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@exercise\n",
+    "def scalar_mult(x : complex, a : Matrix) -> Matrix:\n",
+    "    # Fill in the missing code and run the cell to check your work.\n",
+    "    return # ..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Matrix Multiplication\n",
+    "\n",
+    "A more complicated, but much more important and interesting operation is **matrix multiplication**. The first complicated thing about it is that neither its operands nor its output are the same size: an $n \\times m$ matrix multiplied by an $m \\times k$ matrix results in an $n \\times k$ matrix. That is, for matrix multiplication to be applicable, the number of columns in the first matrix must equal the number of rows in the second matrix.\n",
+    "\n",
+    "Here is how matrix product is calculated: if we are calculating $AB = C$, then\n",
+    "\n",
+    "$$C_{i,j} = A_{i,0} \\cdot B_{0,j} + A_{i,1} \\cdot B_{1,j} + \\dotsb + A_{i,m-1} \\cdot B_{m-1,j} = \\sum_{t = 0}^{m-1} A_{i,t} \\cdot B_{t,j}$$\n",
+    "\n",
+    "Here is an example with relatively small matrices:\n",
+    "\n",
+    "$$\\begin{bmatrix}\n",
+    "    1 & 2 & 3 \\\\\n",
+    "    4 & 5 & 6\n",
+    "\\end{bmatrix}\n",
+    "\\begin{bmatrix}\n",
+    "    1 & 2 \\\\\n",
+    "    3 & 4 \\\\\n",
+    "    5 & 6\n",
+    "\\end{bmatrix}\n",
+    "=\n",
+    "\\begin{bmatrix}\n",
+    "    (1 \\cdot 1) + (2 \\cdot 3) + (3 \\cdot 5) & (1 \\cdot 2) + (2 \\cdot 4) + (3 \\cdot 6) \\\\\n",
+    "    (4 \\cdot 1) + (5 \\cdot 3) + (6 \\cdot 5) & (4 \\cdot 2) + (5 \\cdot 4) + (6 \\cdot 6)\n",
+    "\\end{bmatrix}\n",
+    "=\n",
+    "\\begin{bmatrix}\n",
+    "    22 & 28 \\\\\n",
+    "    49 & 64\n",
+    "\\end{bmatrix}$$\n",
+    "\n",
+    "Note that matrix multiplication is **not commutative:** $AB$ rarely equals $BA$. That being said, here are the properties it does have:\n",
+    "\n",
+    "* Associativity: $A(BC) = (AB)C$\n",
+    "* Distributivity over matrix addition: $A(B + C) = AB + AC$ and $(A + B)C = AC + BC$\n",
+    "* Associativity with scalar multiplication: $xAB = x(AB) = A(xB)$\n",
+    "\n",
+    "Another very important property of matrix multiplication is that a matrix multiplied by a vector produces another vector.\n",
+    "\n",
+    "An **identity matrix** $I_n$ is a special $n \\times n$ matrix which has $1$s on the main diagonal, and $0$s everywhere else:\n",
+    "\n",
+    "$$I_n =\n",
+    "\\begin{bmatrix}\n",
+    "    1 & 0 & 0 & \\dotsb & 0 \\\\\n",
```

Comment:
I'd remove one row and one column from the matrix, it seems enough to have just three rows/columns to get the pattern.